examples/mlperf/model_train.py

```diff
@@ -354,15 +354,22 @@ def train_rnnt():
   pass
 
 @TinyJit
-def train_step_bert(model, optimizer, scheduler, input_ids:Tensor, segment_ids:Tensor, attention_mask:Tensor, masked_positions:Tensor, masked_lm_ids:Tensor, masked_lm_weights:Tensor, next_sentence_labels:Tensor):
+def train_step_bert(model, optimizer, scheduler, input_ids:Tensor, segment_ids:Tensor, attention_mask:Tensor, masked_positions:Tensor, masked_lm_ids:Tensor, masked_lm_weights:Tensor, next_sentence_labels:Tensor, loss_scaler:float):
   lm_logits, clsf_logits = model(input_ids, segment_ids, attention_mask, masked_positions)
   lm_loss = lm_logits.sparse_categorical_crossentropy(masked_lm_ids, ignore_index=masked_lm_weights)
   clsf_loss = clsf_logits.binary_crossentropy_logits(next_sentence_labels)
   loss = lm_loss + clsf_loss
 
   if not getenv('DISABLE_BACKWARD', 0):
     optimizer.zero_grad()
-    loss.backward()
+    (loss * loss_scaler).backward()
+
+    global_norm = Tensor.empty((len(optimizer.params),), device=optimizer[0].device).realize()
+    for i, p in enumerate(optimizer.params): 
+      p.grad = p.grad / loss_scaler
+      global_norm[i] = p.grad.square().sum()
+    global_norm = global_norm.sum().sqrt()
+    for i, p in enumerate(optimizer.params): p.grad = p.grad / Tensor.where(global_norm > 1.0, global_norm, 1.0)
```

Comment:
Isn't `global_norm` just a scalar here? I don't really get the use of `.where` here.

Reply:
@wozeparrot Yeah, it's a scalar. But what other way is there to evaluate the expression, but not break the graph?